### PR TITLE
57 remove redundant csv saving when saving hirds data to database

### PR DESCRIPTION
**Pull request by caseyli92**
DESCRIPTION OF PR:
Currently rainfall data is fetched from the web, saved to CSV, read from CSV, and then saved to the database.
I have checked with Matt, we can remove the redundant CSV saving and just save directly to the database.

Fixes: #57 

## Developer Checklist
- [x] Make code change
- [ ] Update tests
  - [ ] Update / create new tests
  - [ ] Ensure these tests have the expected behaviour
  - [ ] Test locally and ensure tests are passing
- [x] Update documentation
  - [x] Readme
  - [x] Docstrings
  - [x] Comments
  - [ ] Wiki

## Reviewer Checklist
- [ ] Check new code for code smells
- [ ] Check new tests
  - [ ] Ensure adequate coverage
  - [ ] Check for code smells within tests
- [ ] Check if documentation needs updating
  - [ ] Readme
  - [ ] Docstrings
  - [ ] Comments
  - [ ] Wiki

### CODE REVIEW

**Comment by caseyli92** [comment]
Hi @rosepearson and @LukeParky, this is ready for review now :)

File: src/dynamic_boundary_conditions/rainfall_data_from_hirds.py [hunk 1]
```diff
@@ -86,24 +86,59 @@ def get_data_from_hirds(site_id: str, idf: bool) -> str:
...
+    # Read the site_data text string line by line with a for loop
+    for index, line in enumerate(StringIO(site_data)):
+        # Get lines that contain "(mm) ::" for depth data or "(mm/hr) ::" for intensity data
+        if (line.find("(mm) ::") != -1) or (line.find("(mm/hr) ::") != -1):
+            # Add the row number to skip_rows list
+            skip_rows.append(index + 1)
+            # Add the obtained rcp and time_period values to list
+            rcp_result = re.search(r"(\d*\.\d*)", line)
+            period_result = re.search(r"(\d{4}-\d{4})", line)
+            if rcp_result or period_result is not None:
```
Comment (LukeParky) [inline src/dynamic_boundary_conditions/rainfall_data_from_hirds.py:110]: Should this be:
```suggestion
            if rcp_result is not None or period_result is not None:
```
or alternatively
```suggestion
            if any(result is not None for result in [rcp_result, period_result]):
```
Its very unlikely to cause an issue either way but I like to be explicit

**Review by LukeParky — APPROVED** [review]
Good work, a nice concise PR that cleans up some code to make it easier to read as well as possibly making it more efficient.

**Review by rosepearson — APPROVED** [review]
I echo @LukeParky comment. Nice work - this all looks good to me. Great to see the tidy-up and restructure. It'll definitely make things easier for your next issue :)

**Pull request merged**